src/common/utility.c

```diff
@@ -911,10 +911,18 @@ RsvgDimensionData dt_get_svg_dimension(RsvgHandle *svg)
     }
     else
     {
+#define VIEWPORT_SIZE 10000 //some random big value to have enough precision when size is converted to int
```

Comment:
I would recommend 32767, as that is the largest image size supported by Cairo.  (Watermarks are currently broken anyway for full-size exports of images with dimensions greater than that, such as stitched panoramas.)

Reply:
I'm not completely sure what consequences this number has (e.g. on performance and render quality). Probably not a very good idea to actually go to the limit, given that watermark can be rotated... How about 16384?

Reply:
I suppose it depends on how the 'get geometry' function is implemented.  But if the resulting dimensions generate an upscaled image that later needs to be downscaled before adding to the image, then the best viewport size to use is probably the actual image size rather than some fixed value (with the caveat that it needs to be clamped to 32767).

Reply:
I've looked at watermark iop code - cairo surface size is calculated based on processing image size, not watermark dimensions. Returned watermark dimensions are used as is for viewport when rendering to that cairo surface prepended by cairo_scale call, so that watermark would fit. Seems that changing the ratio between scale factor and dimensions (viewport size) doesn't affect the quality (and therefore probably speed too). I think because this is vector graphics, these numbers are only used for calculation of coordinates and doesn't affect how graphics are drawn. No downscaling/upscaling of actual image data is happening because of viewport size. E.g. viewport 1x1 produced visually good result except for positioning (that's what happened in #10685). So the only thing that VIEWPORT_SIZE constant affects is precision of watermark aspect ratio due to conversion between float and int. Given that maximum image size is 32767, the extreme case would be aspect ratio of 1/32767 (though 1 pixel line is probably not very useful as a watermark). So OK, let's go with VIEWPORT_SIZE == 32767, at least it's not a random number.